### PR TITLE
Add sanitized build

### PR DESCRIPTION
**Pull request by rmorozov**
Add initial support for sanitised builds

### CODE REVIEW

File: cmake/sanitizer.cmake [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#if ((NOT ${CMAKE_CXX_COMPILER_ID} MATCHES "Clang") AND (NOT ${CMAKE_CXX_COMPILER_ID} MATCHES "GNU"))
```
Comment (flexferrum) [inline cmake/sanitizer.cmake]: Is it necessary here? Or we can remove it?

Comment (rmorozov) [reply cmake/sanitizer.cmake]: I think I'll extend it to include Mac OS compiler and uncomment it

File: CMakeLists.txt [hunk 2]
```diff
@@ -86,13 +104,42 @@ add_library(${LIB_TARGET_NAME} ${LIB_LINK_TYPE}
 string(TOUPPER "${CMAKE_BUILD_TYPE}" BUILD_CFG_NAME)
 set(CURRENT_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CMAKE_CXX_FLAGS_${BUILD_CFG_NAME}}")
 
+set(JINJA2CPP_PUBLIC_LIBS "${JINJA2CPP_PUBLIC_LIBS}")
```
Comment (flexferrum) [inline CMakeLists.txt]: Are quotes really necessary here? If `JINJA2CPP_PUBLIC_LIBS` is already a list, it will convert it to string.

Comment (rmorozov) [reply CMakeLists.txt]: Lower I append to it using string concat so it is valuable

**Pull request merged**